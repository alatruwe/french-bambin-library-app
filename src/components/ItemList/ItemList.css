.item-list ul {
  list-style-type: none;
}
