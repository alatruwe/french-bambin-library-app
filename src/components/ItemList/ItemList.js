import React from "react";
import Item from "../Item/Item";
import "./ItemList.css";
import data from "../../data.js";

// API call here to get data

export default class ItemList extends React.Component {
  static defaultProps = {};

  render() {
    return (
      <section className="item-list">
        <ul>
          {data.map((item) => (
            <li key={item.id}>
              <Item
                title={item.title}
                image={item.img}
                description={item.description}
              />
            </li>
          ))}
        </ul>
      </section>
    );
  }
}
