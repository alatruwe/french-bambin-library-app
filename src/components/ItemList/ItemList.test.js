import React from "react";
import ReactDOM from "react-dom";
import ItemList from "./ItemList";
import renderer from "react-test-renderer";

describe(`ItemLList component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ItemList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<ItemList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
