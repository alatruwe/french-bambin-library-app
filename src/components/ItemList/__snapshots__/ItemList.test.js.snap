// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemLList component renders the UI as expected 1`] = `
<section
  className="item-list"
>
  <ul>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Sami et Julie à la plage
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/51Griyx+2dL._SX437_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              Munis de leur seau, de leur pelle et leurs brassards, Sami et Julie vont passer un bel après-midi à la plage! Cette collection de lecture à deux voix, dans laquelle l’adulte lit le texte et l’enfant nomme les mots illustrés, permet à votre enfant de structurer son langage, de se préparer à l’apprentissage de la lecture, et de comprendre ses propres émotions grâce à des thèmes qui lui sont familiers. Des petites histoires qui font parler les enfants! 
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Le petit Chaperon rouge
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/41jk5sDc57L._SX412_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              Un livre pour entrer dans la lecture dès la maternelle - les digrammes sont en vert pour signaler les paires de lettres qui produisent un son. - les lettres muettes sont en gris pour repérer les lettres qui ne se prononcent pas. - des capitales pour faciliter la reconnaissance des lettres. - des bulles pour désigner qui parle. - des onomatopées pour un premier niveau de lecture très simple. Dès que l’enfant connaît le son des lettres et des principaux digrammes (OU, AN, ON…), il peut lire son premier livre !
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Le grand imagier de la nature
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/51zSw0xiTZL._SX391_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              Un somptueux imagier pour goûter au plaisir de la découverte de la nature, une merveilleuse encyclopédie visuelle de la nature.Plus de 600 images choisies et ordonnées par des spécialistes afin de répondre à la curiosité insatiable des enfants.De belles illustrations volontairement réalistes amènent l'enfant à reconnaître facilement le monde qui l'entoure.
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Ludanimo
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/41Rm2wqEQBL._AC_.jpg"
              width="250"
            />
            <p>
              Trois jeux pour les tout-petits Jeu de memo Jeu d'équilibre Jeu de parcours A partir de 3 ans De 1 à 4 joueurs
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Mon amour
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/411Zy3sfj2L._SX432_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              Sur chaque double page, Astrid Desbordes a choisi d'opposer en vis-à-vis des situations contraires, qui jouent avec aisance sur les registres du quotidien et de l'imaginaire, de la poésie et de l'humour. Le message de cet irrésistible album est simple et limpide : l'amour est constant et sans condition. Mais c'est dans la toute dernière phrase que se cache le véritable secret : « Je t'aime parce que tu es mon enfant / mais que tu ne seras jamais à moi. »
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              Le Verger, jeu de cartes
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/716H8Kl%2BQmL._AC_SL1500_.jpg"
              width="250"
            />
            <p>
              Jeu coopératif amusant qui apprend aux jeunes enfants à travailler ensemble. En bois et en carton. Recommandé pour les enfants de 3 ans et plus
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              La Fontaine raconte aux enfants
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/51Wh6q7OsVL._SX380_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              Les sept fables les plus célèbres de Jean de La Fontaine, racontées comme des histoires aux plus petits. Pour découvrir les grands classiques à petits pas. Le Corbeau et le Renard, Le Lièvre et la Tortue, La Cigale et la Fourmi, La Grenouille qui se veut faire aussi grosse que le Boeuf, Le Loup et l' Agneau, Le Renard et la Cigogne, Le Lion et le Rat. Un pan du patrimoine littéraire français : les Fables de La Fontaine. Une démarche pédagogique : pour familiariser les plus petits avec les fables, l' auteur a réécrit ces sept histoires (complexes et difficilement accessibles) en prose, avec des mots simples. Le petit plus : à la fin de chaque histoire, on retrouve la fable originale, telle que La Fontaine l' a écrite au XVIIe siècle. Un beau recueil illustré avec des images simples et fortes, par trois illustrateurs qui ont su donner vie aux animaux de La Fontaine.
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
    <li>
      <section>
        <div>
          <div>
            <h2>
              La tour Eiffel se balade à Paris
            </h2>
            <img
              height="250"
              src="https://images-na.ssl-images-amazon.com/images/I/51uutNbF+tL._SX417_BO1,204,203,200_.jpg"
              width="250"
            />
            <p>
              La tour Eiffel revient en album pour une promenade dans Paris ! Léa et Nino participent à une chasse au trésor dans Paris. Mais la ville est grande et les deux enfants ne savent pas où chercher... La tour Eiffel, qui a la bougeotte et des idées, va les accompagner à la découverte des lieux mythiques de la capitale.
            </p>
          </div>
          <button
            type="button"
          >
            Send an request
          </button>
        </div>
      </section>
    </li>
  </ul>
</section>
`;
